### PR TITLE
[MM-50976] - Contact Support to redirect to Zendesk and pre-fill known information

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
Makes changes such that all "Contact support" ctas redirect to zendesk and "Contact sales" ctas redirect to mm.com

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-50976

#### Related Pull Requests
https://github.com/mattermost/customer-web-server/pull/1114

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: utils/contact.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {LicenseLinks} from './constants';
+
+const baseZendeskFormURL = 'https://support.mattermost.com/hc/en-us/requests/new';
+
```
Comment (marianunez) [inline utils/contact.ts:7]: nit: for the file name `contact_support_sales.ts`

File: utils/contact.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+};
+
+export const goToSelfHostedSupportForm = (email: string, subject: string) => {
+    const form = ZendeskSupportForm.SELF_HOSTED_SUPPORT_FORM;
+    const url = buildZendeskSupportForm(form, [
+        {id: ZendeskFormFieldIDs.EMAIL, val: email},
+        {id: ZendeskFormFieldIDs.SUBJECT, val: subject},
+    ]);
+    window.open(url, '_blank');
+};
```
Comment (marianunez) [inline utils/contact.ts:73]: Are these used ?

**Review by marianunez — CHANGES_REQUESTED** [review]
Code LGTM, mostly a comment about sending the email address in the case of self-hosted and tests need to be updated - unit tests and cypress tests are failing https://github.com/mattermost/mattermost-webapp/pull/12304#issuecomment-1451812850

**Comment by yasserfaraazkhan** [comment]
Hi @AGMETEOR ,
was seeing this error on the spinwick,
Screenshot 2023-03-06 at 3 30 17 PM

File: components/admin_console/feature_discovery/feature_discovery.tsx [hunk 3]
```diff
@@ -96,6 +98,15 @@ export default class FeatureDiscovery extends React.PureComponent<Props, State>
         });
     }
 
+    contactSalesFunc = () => {
```
[the remark this answers is not in the document]
Comment (marianunez) [reply components/admin_console/feature_discovery/feature_discovery.tsx:101]: Gotcha!

File: utils/contact.ts (hunk 2 above)
Comment (marianunez) [reply utils/contact.ts]: Couldn't find any reference to `goToBillingGeneralSalesSupportForm`

**Review by marianunez — COMMENTED** [review]
Latest changes look good. Still pending tests to be updated - unit tests and cypress tests are failing https://github.com/mattermost/mattermost-webapp/pull/12304#issuecomment-1451812850

File: components/common/hooks/useOpenSalesLink.ts [hunk 4]
```diff
@@ -3,11 +3,21 @@
...
+import {LicenseLinks} from 'utils/constants';
 
-export default function useOpenSalesLink(issue?: SalesInquiryIssue, inquireType: InquiryType = InquiryType.Sales) {
-    const contactSalesLink = useSelector(getCloudContactUsLink)(inquireType, issue);
+export default function useOpenSalesLink(): [() => void, string] {
+    const customer = useSelector(getCloudCustomer);
+    const customerEmail = customer?.email || '';
+    const firstName = customer?.contact_first_name || '';
+    const lastName = customer?.contact_last_name || '';
+    const companyName = customer?.name || '';
```
[the remark this answers is not in the document]
Comment (marianunez) [reply components/common/hooks/useOpenSalesLink.ts]: Sorry, second question was an implementation detail that Allan already took care of 👍

**Review by marianunez — APPROVED** [review]
Thank @AGMETEOR!

**Comment by yasserfaraazkhan** [comment]
Hi @AGMETEOR @marianunez 
should we wait till this [MM-51018](https://mattermost.atlassian.net/browse/MM-51018) is merged ? Currently, not seeing the form filled when clicked on contact sales

**Review by yasserfaraazkhan — APPROVED** [review]
checked the navigation when clicked on contact sales, from console, pricing modal.
working as expected

**Comment by marianunez** [comment]
> Hi @AGMETEOR @marianunez should we wait till this [MM-51018](https://mattermost.atlassian.net/browse/MM-51018) is merged ? Currently, not seeing the form filled when clicked on contact sales

@emdecr will be working on supporting these params in mattermost.com side. We can still move forward with completing the work here and if this is used before that ticket is complete they will still be redirected to the correct form but without the field pre-populated. I don't think this is a blocker for this change. We can do the full validation of the prefilled info once published in mm.com

**Comment by emdecr** [comment]
Thanks @marianunez 👍 

@yasserfaraazkhan I can send you a URL and creds for our development site if you want to check what the the decode looks like. I'll send a DM. I've got a in-progress branch up there.

**Review by ConorMacpherson — APPROVED** [review]
LGTM! Nice work. I just left two small comments that might require a change.

**Review by nickmisasi — APPROVED** [review]
Sorry for the delay. LGTM

**Pull request closed without merging**